external_parser/event_processors/joined_event.h

```diff
@@ -30,6 +30,7 @@ struct typed_joined_event {
 
   virtual void calculate_metrics(dsjson_metrics*) {}
   virtual float get_sum_original_reward() const = 0;
+  virtual float get_sum_original_reward_first_slot() const = 0;
```

Comment:
this virtual function should not be introduced here since `first slot` is something that isn't applicable to all reductions (cb/cats)

what can be done is write an override for `calculate_metrics` in `ccb_joined_event` and call `calculate_metrics` from example_joiner line 464 instead (now that I look at it `sum_cost_original` should also go in there, but that is outside the scope of this PR)

Reply:
Actually it might be fine to leave it like this, as long as we assume that all reductions have one slot (maybe add a comment above)